WcaOnRails/app/controllers/api/v0/roles_controller.rb

```diff
@@ -145,36 +202,84 @@ def show
   end
 
   def update
-    user_id = params.require(:userId)
-    delegate_status = params.require(:delegateStatus)
-    region_id = params.require(:regionId)
-    location = params.require(:location)
+    id = params.require(:id)
 
-    user = User.find(user_id)
-    if delegate_status == "senior_delegate"
-      senior_delegate_id = nil
-      User.where(region_id: region_id).where.not(id: user_id).update_all(senior_delegate_id: user_id)
+    if id == "dummyRoleId" # dummyRoleId is temporarily reserved for delegate changes.
+      user_id = params.require(:userId)
+      delegate_status = params.require(:delegateStatus)
+      region_id = params.require(:regionId)
+      location = params.require(:location)
+
+      user = User.find(user_id)
+      if delegate_status == "senior_delegate"
+        senior_delegate_id = nil
+        User.where(region_id: region_id).where.not(id: user_id).update_all(senior_delegate_id: user_id)
+      else
+        senior_delegate_id = User.where(delegate_status: "senior_delegate", region_id: region_id).first.id
+      end
+      user.update!(delegate_status: delegate_status, senior_delegate_id: senior_delegate_id, region_id: region_id, location: location)
+      send_role_change_notification(user)
+
+      render json: {
+        success: true,
+      }
```

Comment:
Isn't there another PR waiting and ready that drops `senior_delegate_id`? Guess that one should be merged first, and then you can rebase this to save yourself some `if`s